django/db/models/expressions.py

```diff
@@ -1419,3 +1419,27 @@ class ValueRange(WindowFrame):
 
     def window_frame_start_end(self, connection, start, end):
         return connection.ops.window_frame_range_start_end(start, end)
+
+
+class Collate(Expression):
+    template = "%(expression)s COLLATE %(collation)s"
+
+    def __init__(self, expression, collation):
+        super().__init__(output_field=None)
+        self.expression = expression
+        self.collation = Value(collation)
```

Comment:
No need to resolve `collation` if you remove it from source expressions.

```suggestion
        self.collation = collation
```